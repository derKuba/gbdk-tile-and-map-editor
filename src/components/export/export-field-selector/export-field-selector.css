:host {
  display: block;
}

.field-box {
  display: flex;
  flex-direction: column;
  border: 1px solid black;
  margin: 0 0 25px 15px;
}


.field-box.active {
  border: 5px solid #33ccff;
  width: 40px;
  height: 40px;
}

.matrixRow {
  display: inline-flex;
}

.field {
  display: inline-flex;
  width: 5px;
  height: 5px;
}


.gb-green {
  background-color: #6b790e;
}

.gb-grey {
  background-color: #c5bfbf;
}

.gb-grey-dark {
  background-color: #5a5760;
}

.gb-white {
  background-color: #fff;
}

.gb-black {
  background-color: #000;
}