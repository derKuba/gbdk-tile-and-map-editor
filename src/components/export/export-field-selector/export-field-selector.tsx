import { Component, Host, h, Prop } from '@stencil/core';

@Component({
    tag: 'mvjp-export-field-selector',
    styleUrl: 'export-field-selector.css',
    shadow: false,
})
export class ExportFieldSelector {
    /**
     *  navigate trough the overlay
     */
    @Prop() navigateTo: Function;

    /**
     *  the drawn matrix
     */
    @Prop() matrix: number[][][];

    /**
     *  selected checkboxes
     */
    @Prop() selectedFields: number[];

    /**
     *  handle choosing checkbox event
     */
    @Prop() handleTileClick: Function;

    private handleButtonClick = () => {
        if (this.selectedFields.length === 0) return;
        this.navigateTo();
    };

    render() {
        const field_checkboxes = this.matrix.map((field, position) => {
            return (
                <mvjp-export-checkbox-field
                    field={[...field]}
                    position={position}
                    isActive={this.selectedFields.includes(position)}
                    setSelectedTile={this.handleTileClick}
                    key={`checkbox_id_${position}`}
                />
            );
        });

        return (
            <Host>
                <div class="fields">{field_checkboxes}</div>

                <button onClick={this.handleButtonClick}> Export </button>
            </Host>
        );
    }
}
