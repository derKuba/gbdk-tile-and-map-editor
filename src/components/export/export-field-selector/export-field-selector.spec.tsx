import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { ExportFieldSelector } from './export-field-selector';

describe('mvjp-export-field-selector', () => {
    it('builds', () => {
        expect(new ExportFieldSelector()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const matrix = [
            [
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 0, 0, 0, 0, 0, 0],
                [2, 2, 2, 0, 0, 0, 0, 0],
            ],
        ];
        const page = await newSpecPage({
            components: [ExportFieldSelector],
            template: () => {
                return (
                    <mvjp-export-field-selector
                        navigateTo={() => {}}
                        handleTileClick={() => {}}
                        selectedFields={[0]}
                        matrix={matrix}
                    />
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('should test navigate to', async () => {
        const navigateTo = jest.fn();

        const page = await newSpecPage({
            components: [ExportFieldSelector],
            template: () => {
                return (
                    <mvjp-export-field-selector
                        navigateTo={navigateTo}
                        handleTileClick={() => {}}
                        selectedFields={[0]}
                        matrix={[]}
                    />
                );
            },
        });

        page.root.querySelector('button').click();
        await page.waitForChanges();
        expect(navigateTo).toBeCalled();
    });

    it('should not navigate to', async () => {
        const navigateTo = jest.fn();

        const page = await newSpecPage({
            components: [ExportFieldSelector],
            template: () => {
                return (
                    <mvjp-export-field-selector
                        navigateTo={navigateTo}
                        handleTileClick={() => {}}
                        matrix={[]}
                        selectedFields={[]}
                    />
                );
            },
        });

        page.root.querySelector('button').click();
        await page.waitForChanges();
        expect(navigateTo).not.toBeCalled();
    });
});
