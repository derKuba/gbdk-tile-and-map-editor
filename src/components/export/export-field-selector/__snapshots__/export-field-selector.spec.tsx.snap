// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-export-field-selector matches snapshot 1`] = `
<mvjp-export-field-selector>
  <div class="fields">
    <mvjp-export-checkbox-field isactive="" position="0"></mvjp-export-checkbox-field>
  </div>
  <button>
    Export
  </button>
</mvjp-export-field-selector>
`;
